huntsman/tests/test_states.py

```
@@ -0,0 +1,194 @@
+    return create_dome_from_config(config_with_simulated_stuff)
+
+
+@pytest.fixture(scope='function')
+def mount(config_with_simulated_stuff):
+    return create_mount_from_config(config_with_simulated_stuff)
+
+
+@pytest.fixture(scope='function')
+def observatory(config_with_simulated_stuff, db_type, cameras, scheduler, dome, mount): 
+    observatory = Observatory(config=config_with_simulated_stuff, cameras=cameras,
+                              scheduler=scheduler, dome=dome, mount=mount, db_type=db_type)
+    return observatory
+
+
+@pytest.fixture(scope='function')
+def pocs(config_with_simulated_stuff, observatory):
+    pocs = POCS(observatory, run_once=True, config=config_with_simulated_stuff)
+    yield pocs
+    pocs.power_down()
+
+# ==============================================================================
+
+
+def test_parking_ready(pocs):
+    '''
+    Test if the parking state transitions back into ready.
+    '''
+    pocs.initialize()
+    pocs.get_ready()
+    pocs.config['simulator'] = [_ for _ in pocs.config['simulator'] if _ != 'night']
+    os.environ['POCSTIME'] = '2016-08-13 23:00:00'
+    assert(not pocs.is_dark(horizon='observe'))
+    pocs.next_state = 'parking'
+    for state in ['parking', 'parked', 'housekeeping', 'sleeping', 'ready']:
+        assert(pocs.next_state == state)
+        pocs.goto_next_state()
+        assert(pocs.state == state)
+    pocs.power_down()
+
+
+def test_sleeping_error(pocs):
+    '''
+    Test if ready transitions to sleeping and then errors if still dark.
+    '''
+    pocs.initialize()
+    pocs.get_ready()
+    # Make sure its dark
+    assert(pocs.is_dark(horizon='observe'))
+    # Get into the sleeping state
+    pocs.next_state = 'parking'
+    for state in ['parking', 'parked', 'housekeeping', 'sleeping']:
+        assert(pocs.next_state == state)
+        if state == 'sleeping':
+            with pytest.raises(RuntimeError):
+                pocs.goto_next_state()
```

Why does this throw a `RuntimeError` if it's dark?

---

Because if its not supposed to be in the sleeping state then it should raise an error rather than continuing indefinitely which is what it currently does

---

Might want to use one of the custom errors from POCS utils (or `panoptes-utils`) in that case, e.g. `PanError`.

---

See note below.